addons/website/static/tests/tours/edit_menus.js

```diff
@@ -248,6 +248,11 @@ wTourUtils.registerWebsitePreviewTour('edit_menus', {
                 .scrollIntoView(true);
         }
     },
+    {
+        content: "Check that the mega menu is closed",
+        trigger: 'iframe #top_menu .nav-item:contains("Megaaaaa!"):has(div[data-name="Mega Menu"]:not(.show))',
```

Comment:
Since the testing suite when checking for a trigger selector is actually checking for a **visible** element matching the trigger selector, you could've probably just check for a selector targeting the mega menu content inside a `:not()`.
Since it would be checking for an existing but invisible DOM element, it wouldn't be found (as it's still not closed and still opened) and the step would wait for it to be hidden.
The step would be checking what you really wants to check (that the mega menu is no more visible) instead of relying on the related menu (having a class implying that the related mega menu is not shown).

Something like `iframe #top_menu:not(:has(.class_of_the_mega_menu_itself))`.

But:
- Need to be confirmed
- Useless to do it as you already ensure through multi build that the fix was fine. I just wanted to let you know about the invisible part of the testing suite internal checks.